workflow-templates/test-package.yaml

```diff
@@ -0,0 +1,72 @@
+name: Test
+
+on:
+  # Run CI on pushes to the main branch, and on PRs against main.
+  push:
+    branches: [ $default-branch ]
+  pull_request:
+    branches: [ $default-branch ]
+
+env:
+  PUB_ENVIRONMENT: bot.github
+
+jobs:
+  # Check code formatting and static analysis on a single OS (linux)
+  # against Dart stable and dev.
+  analyze:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        sdk: [dev, stable]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache Pub hosted dependencies
+        uses: actions/cache@v2
+        with:
+          path: "~/.pub-cache/hosted"
+          key: "os:ubuntu-latest;pub-cache-hosted;dart:${{ matrix.sdk }};packages:${{ github.repository }}"
+          restore-keys: |
+            os:ubuntu-latest;pub-cache-hosted;dart:${{ matrix.sdk }};packages:${{ github.repository }}
+            os:ubuntu-latest;pub-cache-hosted;dart:${{ matrix.sdk }}
+            os:ubuntu-latest;pub-cache-hosted
+            os:ubuntu-latest
+      - uses: dart-lang/setup-dart@v0.1
+        with:
+          channel: ${{ matrix.sdk }}
+      - name: Install dependencies
+        run: dart pub get
+      - name: Check formatting
+        run: dart format --output=none --set-exit-if-changed .
+      - name: Analyze code
+        run: dart analyze
+
+  # Run tests on a matrix consisting of three dimensions:
+  # 1. OS: mac, windows, linux
+  # 2. release channel: dev, stable
+  test:
+    needs: analyze
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
```

Comment:
Do we really need/want to run on all OSes for all packages?

Reply:
Seems most packages run on linux by default, so I switched to that and added a comment instead.